### PR TITLE
BatchMatMul_WIP

### PR DESCRIPTION
**Pull request by jianyinglang**
Missing the graph generation in C++ unit test

### CODE REVIEW

File: src/ngraph_stub_ops.cc [hunk 1]
```diff
@@ -60,6 +60,9 @@ REGISTER_NGRAPH_STUB(Name("Add")
 REGISTER_NGRAPH_STUB(Name("AvgPool")
                          .Device(ngraph_bridge::DEVICE_NGRAPH)
                          .TypeConstraint<float>("T"));
+REGISTER_NGRAPH_STUB(Name("BatchMatMul")
+                         .Device(ngraph_bridge::DEVICE_NGRAPH)
```
Comment (avijit-nervana) [inline src/ngraph_stub_ops.cc]: I believe this now goes to the confirmation pass. @aprocter can provide better info.

File: test/tf_exec.cpp [hunk 2]
```diff
@@ -104,4 +104,36 @@ TEST(tf_exec, axpy) {
   }
 }
 
+TEST(tf_exec, BatchMatMul) { 
+  tf::Scope root = tf::Scope::NewRootScope();
+  auto A = tf::ops::Const(root, {-1.f, 2.f, 3.f, 4.f}, tf::TensorShape({2,1,2})); 
+  auto B = tf::ops::Const(root, {1.f, 0.f, -1.f, -2.f}, tf::TensorShape({2,2,1})); 
+  auto R = tf::ops::BatchMatMul(root.WithOpName("R"), A, B);
+  std::vector<tf::Tensor> outputs;
```
Comment (avijit-nervana) [inline test/tf_exec.cpp]: Please use the following code snippet to put the computation to NGRAPH device:
```
  auto dev_scope = root.WithDevice("/device:NGRAPH:0");

  auto A = tf::ops::Const(dev_scope, {-1.f, 2.f, 3.f, 4.f},
                          tf::TensorShape({2, 1, 2}));
  auto B = tf::ops::Const(dev_scope, {1.f, 0.f, -1.f, -2.f},
                          tf::TensorShape({2, 2, 1}));
  auto R = tf::ops::BatchMatMul(dev_scope.WithOpName("R"), A, B);
```

File: test/tf_exec.cpp [hunk 3]
```diff
@@ -104,4 +104,36 @@ TEST(tf_exec, axpy) {
...
+  // Run and fetch v
+  tf::ClientSession session(root);
+  TF_CHECK_OK(session.Run({R}, &outputs));
+  // Expect outputs[0] == [19; -3]
+  ASSERT_EQ(outputs[0].shape(), tf::TensorShape({2,1,1}));
+}
+
+TEST(tf_exec, BatchMatMul_2D) { 
+  tf::Scope root = tf::Scope::NewRootScope();
+  auto A = tf::ops::Const(root, {-1.f, 2.f, 3.f, 4.f}, tf::TensorShape({2,2}));
```
Comment (avijit-nervana) [inline test/tf_exec.cpp]: Same comment as above for placing the computation on NGRAPH

File: src/ngraph_stub_ops.cc (hunk 1 above)
Comment (aprocter) [reply src/ngraph_stub_ops.cc]: This is correct. There are three things you need to do:

(1) register type constraints in the confirmation pass
(2) register a confirmation function in the confirmation pass
(3) REGISTER_NGRAPH_STUB (but this now just takes a "string" argument)

Please check out existing patterns in `master` for details.

**Comment by crlishka** [comment]
Re-running, as the merge job was requesting the wrong tensorflow version for CI builds:

**Comment by crlishka** [comment]
TESTNOW

**Comment by crlishka** [comment]
@jianyinglang  I looked at the failure in the most recent CI log, and I think you may need to merge from _master_.  It looks like you may branched before my PR for updating the CI builds was merged -- merging again from _master_ should fix this.

**Comment by jianyinglang** [comment]
The current branch should pass the unit test

**Pull request closed without merging**